grype/version/gemfile_version_test.go

```diff
@@ -0,0 +1,93 @@
+package version
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_newGemfileVersion(t *testing.T) {
```

Comment:
this should additionally cover cases with valid prereleases (e.g. `-alpha` or `-beta-2`) and metadata (e.g. `+build.12`) as well as permutations of these.